### PR TITLE
SAK-46257 Threaded conversations conversion

### PR DESCRIPTION
**Pull request by adrianfish**
https://sakaiproject.atlassian.net/browse/SAK-46257

### CODE REVIEW

**Comment by adrianfish** [comment]
@bjones86 Sorry, mate. Asking you for another Oracle check ... :)

File: docs/conversion/sakai_23_oracle_conversion.sql [hunk 1]
```diff
@@ -23,3 +23,22 @@ ALTER TABLE rbc_rating ADD order_index NUMBER(1,0) NULL;
 UPDATE rbc_rating r, rbc_criterion_ratings cr SET r.criterion_id = cr.rbc_criterion_id, r.order_index = cr.order_index WHERE cr.ratings_id = r.id;
 UPDATE rbc_criterion c, rbc_rubric_criterions rc SET c.rubric_id = rc.rbc_rubric_id, c.order_index = rc.order_index WHERE rc.criterions_id = c.id;
 -- END SAK-46178
+
+-- SAK-46257
+ALTER TABLE CONV_POSTS ADD DEPTH NUMBER(2, 0) DEFAULT NULL;
+ALTER TABLE CONV_TOPIC_STATUS ADD POSTED CHAR(1) DEFAULT NULL;
```
Comment (bjones86) [inline docs/conversion/sakai_23_oracle_conversion.sql:29]: Is this mean to represent a boolean true/false/0/1? If so it's more appropriate to use `NUMBER(1,0)`

File: docs/conversion/sakai_23_oracle_conversion.sql [hunk 2]
```diff
@@ -23,3 +23,22 @@ ALTER TABLE rbc_rating ADD order_index NUMBER(1,0) NULL;
...
+ALTER TABLE CONV_TOPICS ADD DUE_DATE TIMESTAMP DEFAULT NULL;
+ALTER TABLE CONV_TOPICS ADD HIDE_DATE TIMESTAMP DEFAULT NULL;
+ALTER TABLE CONV_TOPICS ADD LOCK_DATE TIMESTAMP DEFAULT NULL;
+ALTER TABLE CONV_TOPICS ADD SHOW_DATE TIMESTAMP DEFAULT NULL;
+ALTER TABLE CONV_POSTS ADD HOW_ACTIVE NUMBER(2, 0) DEFAULT NULL;
+ALTER TABLE CONV_COMMENTS ADD TOPIC_ID VARCHAR2(36) NOT NULL;
+ALTER TABLE CONV_POSTS ADD NUMBER_OF_THREAD_REACTIONS NUMBER(8, 0) DEFAULT NULL;
+ALTER TABLE CONV_POSTS ADD NUMBER_OF_THREAD_REPLIES NUMBER(8, 0) DEFAULT NULL;
+ALTER TABLE CONV_POSTS ADD PARENT_THREAD_ID VARCHAR2(255) DEFAULT NULL;
+ALTER TABLE CONV_TOPICS ADD MUST_POST_BEFORE_VIEWING CHAR(1) DEFAULT NULL;
```
Comment (bjones86) [inline docs/conversion/sakai_23_oracle_conversion.sql:39]: Same here

**Comment by bjones86** [comment]
Mostly looks fine to me aside from the possible boolean changes.

One thing I would ask is: can we provide meaningful default values for any of these fields? I really dislike  nullable fields if there is a way around it.

File: docs/conversion/sakai_23_oracle_conversion.sql (hunk 1 above)
Comment (adrianfish) [inline docs/conversion/sakai_23_oracle_conversion.sql:29]:
```suggestion
ALTER TABLE CONV_TOPIC_STATUS ADD POSTED NUMBER(1,0);
```

File: docs/conversion/sakai_23_oracle_conversion.sql [hunk 3]
```diff
@@ -23,3 +23,22 @@ ALTER TABLE rbc_rating ADD order_index NUMBER(1,0) NULL;
 UPDATE rbc_rating r, rbc_criterion_ratings cr SET r.criterion_id = cr.rbc_criterion_id, r.order_index = cr.order_index WHERE cr.ratings_id = r.id;
 UPDATE rbc_criterion c, rbc_rubric_criterions rc SET c.rubric_id = rc.rbc_rubric_id, c.order_index = rc.order_index WHERE rc.criterions_id = c.id;
 -- END SAK-46178
+
+-- SAK-46257
+ALTER TABLE CONV_POSTS ADD DEPTH NUMBER(2, 0) DEFAULT NULL;
```
Comment (adrianfish) [inline docs/conversion/sakai_23_oracle_conversion.sql:28]:
```suggestion
ALTER TABLE CONV_POSTS ADD DEPTH NUMBER(2, 0);
```

File: docs/conversion/sakai_23_oracle_conversion.sql [hunk 4]
```diff
@@ -23,3 +23,22 @@ ALTER TABLE rbc_rating ADD order_index NUMBER(1,0) NULL;
...
+ALTER TABLE CONV_TOPICS ADD DUE_DATE TIMESTAMP DEFAULT NULL;
+ALTER TABLE CONV_TOPICS ADD HIDE_DATE TIMESTAMP DEFAULT NULL;
+ALTER TABLE CONV_TOPICS ADD LOCK_DATE TIMESTAMP DEFAULT NULL;
+ALTER TABLE CONV_TOPICS ADD SHOW_DATE TIMESTAMP DEFAULT NULL;
+ALTER TABLE CONV_POSTS ADD HOW_ACTIVE NUMBER(2, 0);
+ALTER TABLE CONV_COMMENTS ADD TOPIC_ID VARCHAR2(36) NOT NULL;
+ALTER TABLE CONV_POSTS ADD NUMBER_OF_THREAD_REACTIONS NUMBER(8, 0) DEFAULT NULL;
+ALTER TABLE CONV_POSTS ADD NUMBER_OF_THREAD_REPLIES NUMBER(8, 0) DEFAULT NULL;
+ALTER TABLE CONV_POSTS ADD PARENT_THREAD_ID VARCHAR2(255) DEFAULT NULL;
+ALTER TABLE CONV_TOPICS ADD MUST_POST_BEFORE_VIEWING CHAR(1) DEFAULT NULL;
```
Comment (adrianfish) [inline docs/conversion/sakai_23_oracle_conversion.sql:39]:
```suggestion
ALTER TABLE CONV_TOPICS ADD MUST_POST_BEFORE_VIEWING NUMBER(1,0);
```

**Comment by ern** [comment]
these were added in 22.1 scripts so no need to add to 23

**Pull request closed without merging**